### PR TITLE
Story/21: MetadataType description suggestions

### PR DESCRIPTION
**Pull request by eliasvagan**
Update: Ready for review

### CODE REVIEW

**Comment by eliasvagan** [comment]
Resolved merge conflicts, please check to see if I broke something.

**Comment by petterroea** [comment]
Backend does not build

**Comment by Letaguator** [comment]
It builds fine, just a test from an old version of development. Since we started merging when development had tests that failed.

image

**Comment by Letaguator** [comment]
But the front end does not compile however.

**Comment by eliasvagan** [comment]
Currently waiting for #224 to be merged so we can rebase this branch onto development.

File: frontend/src/pages/MetadataByType/MetadataByTypeResults.jsx [hunk 1]
```diff
@@ -100,19 +94,25 @@ const MetadataByTypeResults = ({ metadataTypeUuid }) => {
...
+  }
+  const { tags, description, name } = metadataType;
   return (
     <CategoriesContainer>
       <ResultsHeader>
         <div>
-          <h3>{metadataTypeName}</h3>
-          <p>{description}</p>
+          <h3>{metadataType.name}</h3>
+          <DescriptionEditButton uuid={metadataTypeUuid} currentDescription={description.content} />
```
Comment (MittBrukernavn) [inline frontend/src/pages/MetadataByType/MetadataByTypeResults.jsx:114]: DescriptionEditButton has no prop currentDescription

```suggestion
          <DescriptionEditButton uuid={metadataTypeUuid} />
```

Comment (MittBrukernavn) [inline frontend/src/pages/MetadataByType/MetadataByTypeResults.jsx:114]: Also, this button should only be visible to logged in users (and probably only municipality users)

File: frontend/src/router/router.jsx [hunk 2]
```diff
@@ -25,13 +25,15 @@ import { Search } from '../pages/Search';
...
 import { Category } from '../pages/Category';
+import { EditDescriptionMetadataType } from '../pages/EditDescriptionMetadataType';
 
 const RouterComponent = () => { // eslint-disable-line arrow-body-style
   return (
     <Router history={history}>
       <Switch>
         <Route exact path="/" component={Splash} />
-        <Route path="/dataType/:name" component={MetadataByType} />
+        <PrivateRoute path="/dataType/description/:id" component={EditDescriptionMetadataType} />
```
Comment (MittBrukernavn) [inline frontend/src/router/router.jsx:35]: Should be municipality only. Submitting or voting for description as non-municipality user returns a 401 from the backend.

File: frontend/src/sharedComponents/Metadata/ContentComps.jsx [hunk 3]
```diff
@@ -121,8 +121,31 @@ const StarEmptyStyled = styled(StarEmpty)`
...
+        if (res.status === 200) {
+          const { name } = await res.json();
+          setMetadataTypeName(name);
+        }
+      } catch (err) {
+        const { status } = err;
+        if (status === 404) {
+          dispatch(alertActions.error('Could not fetch metadata type name'));
+        } else if (status === 401) {
+          dispatch(alertActions.info('Not authorized to see metadata type name.'));
```
Comment (MittBrukernavn) [inline frontend/src/sharedComponents/Metadata/ContentComps.jsx:141]: Can this happen?

File: frontend/src/sharedComponents/Metadata/ContentComps.jsx [hunk 4]
```diff
@@ -121,8 +121,31 @@ const StarEmptyStyled = styled(StarEmpty)`
...
+        } else {
+          dispatch(alertActions.error('Failed to fetch metadata type name. Please try again later.'));
+        }
+      }
+    };
+    if (showCategory) getMetadataTypeName();
```
Comment (MittBrukernavn) [inline frontend/src/sharedComponents/Metadata/ContentComps.jsx:147]: Pls
```suggestion
    if (showCategory) {
      getMetadataTypeName();
    }
```

File: frontend/src/sharedComponents/Metadata/ContentComps.jsx [hunk 5]
```diff
@@ -198,6 +221,25 @@ const ContentExpanded = ({ metadata, showCategory, showMunicipality }) => {
...
+        if (res.status === 200) {
+          const { name } = await res.json();
+          setMetadataTypeName(name);
+        }
+      } catch (err) {
+        const { status } = err;
+        if (status === 404) {
+          dispatch(alertActions.error('Could not fetch metadata type name'));
+        } else if (status === 401) {
+          dispatch(alertActions.info('Not authorized to see metadata type name.'));
```
Comment (MittBrukernavn) [inline frontend/src/sharedComponents/Metadata/ContentComps.jsx:236]: Can this happen?

File: frontend/src/sharedComponents/Metadata/ContentComps.jsx [hunk 6]
```diff
@@ -198,6 +221,25 @@ const ContentExpanded = ({ metadata, showCategory, showMunicipality }) => {
...
+      try {
+        const res = await fetch(`/api/MetadataType/${metadataTypeUuid}`);
+        if (res.status === 200) {
+          const { name } = await res.json();
+          setMetadataTypeName(name);
+        }
+      } catch (err) {
+        const { status } = err;
+        if (status === 404) {
+          dispatch(alertActions.error('Could not fetch metadata type name'));
```
Comment (MittBrukernavn) [inline frontend/src/sharedComponents/Metadata/ContentComps.jsx:234]: Could not find may be more specific in this case. Could not fetch indicates something else might have happened in the chain.
```suggestion
          dispatch(alertActions.error('Could not find metadata type name'));
```

File: frontend/src/sharedComponents/Metadata/ContentComps.jsx [hunk 7]
```diff
@@ -198,6 +221,25 @@ const ContentExpanded = ({ metadata, showCategory, showMunicipality }) => {
...
+        } else if (status === 401) {
+          dispatch(alertActions.info('Not authorized to see metadata type name.'));
+        } else {
+          dispatch(alertActions.error('Failed to fetch metadata type name. Please try again later.'));
+        }
+      }
+    };
+    if (showCategory) getMetadataTypeName();
```
Comment (MittBrukernavn) [inline frontend/src/sharedComponents/Metadata/ContentComps.jsx:242]: Pls
```suggestion
    if (showCategory) {
      getMetadataTypeName();
    }
```

File: frontend/src/sharedComponents/Metadata/ContentComps.jsx [hunk 8]
```diff
@@ -121,8 +121,31 @@ const StarEmptyStyled = styled(StarEmpty)`
...
+      try {
+        const res = await fetch(`/api/MetadataType/${metadataTypeUuid}`);
+        if (res.status === 200) {
+          const { name } = await res.json();
+          setMetadataTypeName(name);
+        }
+      } catch (err) {
+        const { status } = err;
+        if (status === 404) {
+          dispatch(alertActions.error('Could not fetch metadata type name'));
```
Comment (MittBrukernavn) [inline frontend/src/sharedComponents/Metadata/ContentComps.jsx:139]: Could not find may be more specific in this case. Could not fetch indicates something else might have happened in the chain.
```suggestion
          dispatch(alertActions.error('Could not find metadata type name'));
```

File: frontend/src/pages/MetadataByType/MetadataByTypeResults.jsx (hunk 1 above)
Comment (eliasvagan) [reply frontend/src/pages/MetadataByType/MetadataByTypeResults.jsx:114]: Good catch

**Pull request merged**